### ISSUE TITLE
Init-pants can surprisingly change order of resolution due to HOME/.local/bin addition

### ISSUE THREAD

**Issue by akomisarek** [pantsbuild/actions#31]
Init-pants can lead to unexpected version of Python/scripts being taken from bin

Let me explain the setup I have encountered:

1. Not sure why but the image we were using had `pip` inside the `/home/runner/.local/bin` 
2. We invoke setup-python before initialising pants. This adds new Python to the PATH. Everything works as expected
3. Now invoking init-pants performed the following: https://github.com/pantsbuild/actions/blob/d767b356a10c6c50fc0db0ddc215611b300b226b/init-pants/action.yaml#L110-L112
This adds /home/runner/.local/bin again to the beginning of the list. 

This lead to wrong pip resolution as the one from ./local/bin is used instead of the newly installed one.

Expected behaviour would be not to modify PATH if .local/bin is already on the path maybe? 

I am also questioning our ./local/bin having Pip but it's different story! 

Thoughts?

**Comment by benjyw** [comment]
Hmmm, one solution is to move `pants` out of `~/.local/bin` to some pants-specfic subdirectory, and put that on the path.

**Comment by akomisarek** [comment]
Thanks again, Ben! :) I did the workaround that I am doing our `setup-python` step after `init-pants` and that also work. But maybe could pick this up in future if not fixed and have got couple of minutes of spare time.

**Comment by kaos** [comment]
yeah, I think the way to go could be to first check if `pants` is discoverable on PATH, and only amend PATH if not.
